[CI] SmokeTestWatcherWithSecurityIT.testSearchInputWithInsufficientPrivileges Failure

*Original comment by @martijnvg:*

A different failure than in LINK REDACTED.

I was unable to reproduce this failure locally.

```
FAILURE 7.72s | SmokeTestWatcherWithSecurityIT.testSearchInputWithInsufficientPrivileges <<< FAILURES!
   > Throwable LINK REDACTED: java.lang.AssertionError: 
   > Expected: is "execution_not_needed"
   >      but: was "not_executed_already_queued"
   > 	at __randomizedtesting.SeedInfo.seed([B75EC64E2E3C5769:18369ABB27EDF467]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.smoketest.SmokeTestWatcherWithSecurityIT.testSearchInputWithInsufficientPrivileges(SmokeTestWatcherWithSecurityIT.java:160)
   > 	at java.lang.Thread.run(Thread.java:748)
```

LINK REDACTED

Build url: LINK REDACTED

### Copy of the relevant failure

<details>
<summary>Expected: is "execution_not_needed" but: was "not_executed_already_queued"</summary>

Stacktrace

java.lang.AssertionError: 
Expected: is "execution_not_needed"
     but: was "not_executed_already_queued"
	at __randomizedtesting.SeedInfo.seed([8C6B46A3AC22313C:23031A56A5F39232]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.smoketest.SmokeTestWatcherWithSecurityIT.testSearchInputWithInsufficientPrivileges(SmokeTestWatcherWithSecurityIT.java:184)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:834)

Standard Output

[2019-05-31T07:04:45,233][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchInputWithInsufficientPrivileges] before test
[2019-05-31T07:04:45,437][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchInputWithInsufficientPrivileges] initializing REST clients against [http://[::1]:36713]
[2019-05-31T07:04:51,872][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchInputWithInsufficientPrivileges] after test
[2019-05-31T07:04:51,920][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchTransformInsufficientPermissions] before test
[2019-05-31T07:04:55,001][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchTransformInsufficientPermissions] after test
[2019-05-31T07:04:55,018][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testIndexActionInsufficientPrivileges] before test
[2019-05-31T07:04:58,013][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testIndexActionInsufficientPrivileges] after test
[2019-05-31T07:04:58,029][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchInputHasPermissions] before test
[2019-05-31T07:05:00,990][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchInputHasPermissions] after test
[2019-05-31T07:05:01,007][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchTransformHasPermissions] before test
[2019-05-31T07:05:03,976][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testSearchTransformHasPermissions] after test
[2019-05-31T07:05:03,991][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testIndexActionHasPermissions] before test
[2019-05-31T07:05:06,897][INFO ][o.e.s.SmokeTestWatcherWithSecurityIT] [testIndexActionHasPermissions] after test

</details>

### Copy of the reproduce line
```
./gradlew :x-pack:qa:smoke-test-watcher-with-security:integTestRunner --tests "org.elasticsearch.smoketest.SmokeTestWatcherWithSecurityIT.testSearchInputWithInsufficientPrivileges" -Dtests.seed=8C6B46A3AC22313C -Dtests.security.manager=true -Dtests.locale=ha -Dtests.timezone=Asia/Omsk -Dcompiler.java=12 -Druntime.java=11
```
(does not reproduce)
### The Jenkins build link
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3818/testReport/junit/org.elasticsearch.smoketest/SmokeTestWatcherWithSecurityIT/testSearchInputWithInsufficientPrivileges/
### The Gradle scan link
https://gradle.com/s/kgzehspj7rn22
### The relevant cluster logs from "Google Cloud Storage Upload Report" (link found in Jenkins build)
Link to full logs: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3818/gcsObjects
Server log file for this failure: [x-pack_qa_smoke-test-watcher-with-security_integTestCluster.log](https://github.com/elastic/elasticsearch/files/3241768/x-pack_qa_smoke-test-watcher-with-security_integTestCluster.log)

--- a/x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java
+++ b/x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java
@@ -164,7 +164,7 @@ public class SmokeTestWatcherWithSecurityIT extends ESRestTestCase {
         String indexName = "index_not_allowed_to_read";
         try (XContentBuilder builder = jsonBuilder()) {
             builder.startObject();
-            builder.startObject("trigger").startObject("schedule").field("interval", "1s").endObject().endObject();
+            builder.startObject("trigger").startObject("schedule").field("interval", "4s").endObject().endObject();
             builder.startObject("input").startObject("search").startObject("request")
                     .startArray("indices").value(indexName).endArray()
                     .startObject("body").startObject("query").startObject("match_all").endObject().endObject().endObject()
@@ -180,8 +180,10 @@ public class SmokeTestWatcherWithSecurityIT extends ESRestTestCase {
 
         // check history, after watch has fired
         ObjectPath objectPath = getWatchHistoryEntry(watchId);
-        String state = objectPath.evaluate("hits.hits.0._source.state");
-        assertThat(state, is("execution_not_needed"));
+        assertBusy(() -> {
+            String state = objectPath.evaluate("hits.hits.0._source.state");
+            assertThat(state, is("execution_not_needed"));
+        });
         boolean conditionMet = objectPath.evaluate("hits.hits.0._source.result.condition.met");
         assertThat(conditionMet, is(false));
     }

